### PR TITLE
Remove sentience from clean and medi bot

### PR DESCRIPTION
**Pull request by VasilisThePikachu**
## Why / Balance

There is no point in these silicon being sentient, they have basically zero role play potential and are not meant to be controlled by players. The janibot can be played somewhat but at that point we have janitor borgs which are way better at this. You can only clean floors and even then it does a terrible job at doing that and only that. A player playing as a janibot will get bored quickly. No amount of RP will save you.

A player taking over a mediborg just makes it useless as you cant inject anymore. And again, medical borg. There's no point in adding the feature. It's too much work then its worth when we have borgs.

They don't have ghost role info for a reason. They are not meant to be played. Other NPC's here don't have sentience either.

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

**Changelog**

:cl: Myra
- remove: Medibots and Janibots can no longer become sentient via the sentience event.

### CODE REVIEW

**Comment by Ilya246** [comment]
removing it from medibots until they're fixed makes sense, but i disagree with the janibot bit - you haven't provided a real reason to remove it from them, and if a player gets bored as one, they can simply just /ghost anytime

**Comment by Simyon264** [comment]
>  And again, medical borg. 

As a cyborg your laws force you to interact with other players more than a medibot. If all you want is to heal people a medibot is better than a cyborg where someone can go "law 2 grab a generator" (as an example)

**Comment by slarticodefast** [comment]
There is a PR for fixing the sentient medibot #32110 so no need to remove it.
Also if someone does not like the ghost role they don't have to take it.

**Comment by UbaserB** [comment]
As a janitor main i can tell you that i wouldnt mind taking the bot to play as

**Comment by Simyon264** [comment]
Also, IIRC medibots (along with others) are playable on some SS13 codebases.

**Comment by VasilisThePikachu** [comment]
> There is a PR for fixing the sentient medibot #32110 so no need to remove it. Also if someone does not like the ghost role they don't have to take it. The ghost role info can easily be added.

It needs a ghost role info

**Comment by BombasterDS** [comment]
> > There is a PR for fixing the sentient medibot #32110 so no need to remove it. Also if someone does not like the ghost role they don't have to take it. The ghost role info can easily be added.
> 
> It needs ghost role info. But i still find it too boring. If a player has to ghost out its not a fun ghost role.

Sometimes you just want to take the role, clean some puddles and leave the role. Without any interactions with players, just to help station without making problems to anyone. It's maybe boring for someone, but for me it's just relaxing.

**Comment by Golinth** [comment]
> A player playing as a janibot will get bored quickly. 

Please speak for yourself. If anything, it’s MORE  interesting than other sentience ghost roles. Using the “it’s boring and has nothing to do, so we should remove it” logic, we shouldn’t have ANY goofy ghost roles. Playing as a mouse you can’t do anything, players will get bored, delete them. Same with snails, vending machines, literally every single other item in the sentience event, plants, pAIs, supply bots, honk bots, and any others I can’t think of rn. They are all limited ghost roles dedicated around RP, that can get boring quick. That doesn’t mean they should be removed from the game.

**Comment by PJB3005** [comment]
My opinion is that this feature is laughably poorly implemented and should not have passed even basic QA. A list of issues that need fixing, in my opinion, before this can be "re-enabled":

* Ghost role text is missing.
* The medibot didn't even work when this was implemented just god damnit man, QA.
* Preferably a way to kick players *out* of a sentient cleanbot, so I can get my cleanbot to do work again if it starts being silly.
* (related to above) a player disconnecting while in a cleanbot just bricks it.
* Need a visible HUD way for players to go "i'm fucking bored" and observe again. i.e. a way to `ghost` without needing to know there's a command for it.
* At least let them be able to make ping sounds or something. Seriously. I'm pretty sure that one Black Mirror episode was better than this crap.

**Comment by Simyon264** [comment]
I agree that they are implemented poorly, but removing the ghost role sucks more than just keeping it in as it is currently. Removing it saying "go fix it" will probably result in no one doing it. Vehicles were removed to rework them and they are still not implemented upstream.

**Comment by PJB3005** [comment]
> I agree that they are implemented poorly, but removing the ghost role sucks more than just keeping it in as it is currently. Removing it saying "go fix it" will probably result in no one doing it.

I am permanently of the opinion that much of the "content" in the game is appallingly low-quality and broken. A half-broken ghost role that didn't get any development past "it's so easy to add 4 lines to a YAML file" and didn't even *work* at first is not something I want to see in this game.

**Comment by VasilisThePikachu** [comment]
> Removing it saying "go fix it" will probably result in no one doing it. Vehicles were removed to rework them and they are still not implemented upstream.

And the same thing will happen if we keep it other then having bad code on the repo. No one wins

**Comment by Simyon264** [comment]
> > Removing it saying "go fix it" will probably result in no one doing it. Vehicles were removed to rework them and they are still not implemented upstream.
> 
> And the same thing will happen if we keep it other then having bad code on the repo. No one wins

See #32110, as mentioned previously by @slarticodefast, some fixes are already being done.

**Comment by Ilya246** [comment]
reminder that temporary changes without concrete plans to reverse them are permanent changes

either way,

> Preferably a way to kick players out of a sentient bot, so I can get my cleanbot to do work again if it starts being silly.

i don't think this is needed, machines malfunctioning occasionally should be part of the gameplay

> A visible way for players to tell that a bot is sentient and being controlled by a player. Like changing the light on it to be a different color. I don't want to start getting bugreports about "my cleanbot doesn't work properly".

i think it adds to roleplay if you can't tell whether the bot is sentient, you can pretend to not be sentient which can be used for funnies; i don't think the point about bugreports is reasonable either

**Comment by slarticodefast** [comment]
> At least let them be able to make ping sounds or something. Seriously. I'm pretty sure that one Black Mirror episode was better than this crap.

There is also a PR giving bots the borg emotes #29949

**Comment by LeIronGoat** [comment]
I agree with potentially removing things because they're broken. (I'm very much one of those developers like many of us who will just leave things if they're "working").

I do disagree with removing (some) things though for being "boring". There's indeed times where I like to just take just such a basic role and interact with people. It can present fun moments. And in this case where its really not that common and is always part of something else gaining sentience, I think its fine if its one of the roles that is "boring".  Sometimes something simple is enjoyable. Personally, i'd also rather play a cleanbot or medibot over a mouse, or mothroach, or personal AI, or sentient vending machine, or a hydroponics tray, or hamlet, or librarian. (okay joking on that last one... kind of) But yes, too many "boring" roles, especially if they block out the more "fun" and "engaging" roles is a bad thing. I think thats about balancing events though (and continuation of adding content.)

That all being said,

I agree with PJB that:
- Ghost role text needs to be added
- A 'ghost' action should be included (though should not be instant - make it an action outside of the action bar, or make it involve a pop-up.) And honestly, this should be included in other ghost roles too like vending machines, hydroponic trays, etc.

I disagree with 
- A way to kick them out. My personal opinion is just destroy it and make another. They're very weak and easy to make.

I can go either way on the highlighting sentience portion. At the end of the day it doesn't matter, the bots have nothing to be stealthy about so its not like it matters (from the bots perspective) if people know they are.

TL;DR
Needs fixes & improvements, but in my opinion still a better ghost-role than a bunch of other ghost-roles.

And yes, I have a slight bias after being a sentient artifact recently who became best friends with a medibot.

**Comment by nymhole** [comment]
extremely lame PR, removes chaos and possibilities from rounds. these bots are not typically sentient and being sentient as one of these bots is not the same as being a borg in any respect. the medibot needs to have a playerUI so that they know who the tricordazine will work on and they need a way to inject tricor, but to remove sentience from both the cleanbot and medibot with no plans to add it back in the future and no plans to ever work on the player controlled medibot is just boring. plain and simple an awful PR.

**Comment by ArtisticRoomba** [comment]
I agree in that medibot sentience as a ghost role was generally half-baked from the start. The thing doesn't even have _Medical_ accesses, so the person playing the role kinda just awkwardly stands at the door waiting for someone to open it for them.

I disagree that these roles have no roleplay potential out of them. I distinctly remember a sentient medibot that would run around Medical and look at patients and check if they were "okay", usually being very witty about it, because they can't do anything to help. When a power outage came Medical gathered around the Medibot in front of a candle as it told a dark story about a Medibot who's core had a bomb strapped to it, only for it to run away when the power came back on, and get stopped by a door that it couldn't open.

If we want the medibot to be a proper mildly useful ghost role, it should probably get:

- [ ] A proper way to use it's abilities (#32110) 
- [ ] Relevant medical accesses
- [ ] Medical comms to draw a doctor's attention to a patient. It only has local and binary right now so this will be interesting in terms of #32385
- [ ] A way to scan patients to become mildly useful in helping diagnosing patients/assisting them, and relaying that info to doctors. Maybe just a medhud as a compromise if we don't want a full fledged scanner. Medibots automatically inject based off of crit/consciousness and don't inject if a patient still has tricord in them so this makes partial sense
- [ ] Statuses of the actual medibot for QOL and consistency (similar to the eternally open #29949)

I understand that this is just toeing the line closer and closer to a Mediborg but having Diet Dr. Tricord the Talking Medibot is funny.

There are many different ways to improve the Medibot even further but that's for way later.

**Comment by Graded5** [comment]
These ghost roles might not have much content to them, but there are still people willing to play them. Why remove them? (Assuming they arent bugged like the medi bot)

Not every ghost role has to be an amazing new way to play the game. Clean bot for example is a good time waster if there isnt any other roles available. Instead of removing them, why dont we just give them more player content? In 13, clean bots can clean dirt of tiles.

**Comment by nymhole** [comment]
>*I don't want to start getting bugreports about "my cleanbot doesn't work properly".
this has been a feature for a while and these bug reports haven't happened. you are responding as if this is a pr for making them sentient.

> * (related to above) a player disconnecting while in a cleanbot just bricks it.
this is a problem for many if not most ghost roles. zombies is a good example. this issue should be rectified directly instead of removing roles like zombies from ghost roles.

**Comment by G-370** [comment]
> removing it from medibots until they're fixed makes sense, but i disagree with the janibot bit - you haven't provided a real reason to remove it from them, and if a player gets bored as one, they can simply just /ghost anytime

Oh fuck! So I didn't need to /suicide ;---;

**Comment by ArtisticRoomba** [comment]
Janibot AI breaks when you ghost out IIRC.
